ACPs/103-dynamic-fees/README.md

```diff
@@ -0,0 +1,133 @@
+
+The X-Chain and P-Chain currently operate under a fixed fee mechanism. To better account for increased load that may result from [ACP-77](../77-reinventing-subnets/README.md), they should be migrated to a dynamic fee mechanism.
+
+## Specification
+
+### Dimensions
+
+There are four dimensions that will be used to approximate the computational cost ("gas" usage) of a transaction:
+
+1. `Bandwidth` ($B$) is the amount of network bandwidth used in transaction broadcast. This is set to the size of the transaction in bytes.
+2. `Reads` ($R$) is the number of state/database reads used in transaction execution.
+3. `Writes` ($W$) is the number of state/database writes used in transaction execution.
+4. `Compute` ($C$) is the total amount of compute used in transaction execution. This is set to the number of signatures verified during transaction execution.
+
+For each transaction, gas ($G$) can be computed (TODO: The weights for each dimension must be specified prior to this ACP being considered "Implementable"):
+
+$$G = B + R + W + C$$
+
+A future ACP should remove the gas standardization to granularly meter usage of each resource in a multidimensional scheme.
+
+### Mechanism
+
+This mechanism aims to have a target gas usage ($T$) per second and adjusts based on the excess gas usage ($x$), defined as the difference between the current gas usage and $T$.
+
+At the start of building/executing block $b$, $x$ is updated:
+
+$$x = \max(x - (T \cdot \Delta t), 0)$$
+
+Where $\Delta t$ is the number of seconds between $b$ and $b$'s parent block.
+
+The required fee per gas for block $b$ is:
+
+$$M \cdot \exp\left(\frac{x}{K}\right)$$
+
+Where:
+
+- $M$ is the minimum gas price
+- $\exp\left(x\right)$ is an approximation of $e^x$
+- $K$ is a constant to control the rate of change for gas price
+
+After processing block $b$, $x$ is updated with the total gas in the block $G$:
+
+$$x = x + G$$
+
+Whenever $x$ increases by $K$, the gas price increases by a factor of `~2.7`. If the gas price gets too expensive, average usage drops, and $x$ starts decreasing, automatically dropping the price again. The gas price constantly adjusts to make sure that, on average, the blockchain uses $T$ gas per second.
+
+$L$ is defined as the maximum gas per second that the blockchain has capacity for. For monotonically increasing blocktimes (as on P-Chain and X-Chain), there may be multiple blocks issued at the same on-chain timestamp. $L$ limits the amount of gas that can be used at $t$, placing a bound on the amount that the required fee can increase from $t$ to $t'$. Once the chain has used $L$ amount of gas at time $t$, a new block will only be considered valid if its timestamp is at least $t + 1$.
```

Comment:
I'm not sure that this is really the mechanism we had talked about (offline). I was under the impression we would be specifying a moving capacity, not a capacity at a specific timestamp.

Reply:
adjusted the phrasing